File: AndroidApp/app/src/main/java/com/malka/androidappp/helper/widgets/DatePickerFragment.java
package com.malka.androidappp.helper.widgets;

import android.app.Activity;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;

import androidx.fragment.app.DialogFragment;

import java.util.Calendar;
import java.util.Date;

public class DatePickerFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener {
    public interface DatePickerListener {
        public void onDateSet(int year, int month, int day);
    }

    private DatePickerListener listener;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the current date as the default date in the picker
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);

        // Create a new instance of DatePickerDialog and return it
        DatePickerDialog dialog = new DatePickerDialog(getActivity(), this, year, month, day);

        dialog.getDatePicker().setMinDate(new Date().getTime());
        return dialog;
    }

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);

        // This makes sure that the container activity has implemented
        // the callback interface. If not, it throws an exception
        try {
            listener = (DatePickerListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString()
                    + " must implement OnHeadlineSelectedListener");
        }
    }

    public void onDateSet(DatePicker view, int year, int month, int day) {
        listener.onDateSet(year, month, day);
    }
}

File: AndroidApp/app/src/main/java/com/malka/androidappp/helper/widgets/DatePickerFragment.kt
package com.malka.androidappp.helper.widgets

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import androidx.fragment.app.DialogFragment
import java.util.*
import kotlin.Int
import kotlin.Unit

class DatePickerFragment(val maxdayToday:Boolean=false,val minDateToday:Boolean=false,val onItemClick:
    (selectdate: String) -> Unit) :
    DialogFragment(), OnDateSetListener {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val c = Calendar.getInstance()
        val year = c[Calendar.YEAR]
        val month = c[Calendar.MONTH]
        val day = c[Calendar.DAY_OF_MONTH]
        val dialog = DatePickerDialog(requireActivity(), this, year, month, day)
        if(maxdayToday){
            dialog.datePicker.maxDate = Date().time
        }
        if(minDateToday){
            dialog.datePicker.minDate = Date().time
        }
        return dialog
    }


    override fun onDateSet(view: DatePicker, year: Int, month: Int, day: Int) {
        var monthString = month.toString()
        if (monthString.length == 1) {
            monthString = "0$monthString"
        }

        val selectdate = "$day/${monthString}/$year"
        onItemClick.invoke(selectdate)
    }
}